hadoop-ozone/recon/src/test/java/org/apache/hadoop/ozone/recon/tasks/TestNSSummaryUnifiedControl.java

```diff
@@ -183,215 +264,300 @@ void testInitialState() {
+      if (attempt == 1) {
+        firstAttempt.countDown();
+        throw new IOException("First failure");
+      } else {
+        secondAttempt.countDown();
+        return null;
+      }
+    }).when(mockNamespaceSummaryManager).clearNSSummaryTable();
+
     // First rebuild fails
-    doThrow(new IOException("Test failure")).when(mockNamespaceSummaryManager).clearNSSummaryTable();
-    
-    TaskResult failedResult = nsSummaryTask.reprocess(mockOMMetadataManager);
-    assertFalse(failedResult.isTaskSuccess(), "First rebuild should fail");
+    ReconTaskController.ReInitializationResult result1 =
+        taskController.queueReInitializationEvent(
+            ReconTaskReInitializationEvent.ReInitializationReason.MANUAL_TRIGGER);
+    assertEquals(ReconTaskController.ReInitializationResult.SUCCESS, result1,
+        "First event should be queued successfully");
+
+    assertTrue(firstAttempt.await(10, TimeUnit.SECONDS), "First rebuild should be attempted");
+    Thread.sleep(1000);
     assertEquals(RebuildState.FAILED, NSSummaryTask.getRebuildState(),
         "State should be FAILED after first rebuild");
 
-    // Second rebuild succeeds
-    // Setup successful rebuild by default - no exception thrown
-    doNothing().when(mockNamespaceSummaryManager).clearNSSummaryTable();
-    
-    TaskResult successResult = nsSummaryTask.reprocess(mockOMMetadataManager);
-    assertTrue(successResult.isTaskSuccess(), "Second rebuild should succeed");
+    // Second rebuild succeeds - wait for retry delay (2 seconds)
+    Thread.sleep(2100); // Wait for RETRY_DELAY_MS (2000ms) + buffer
+
+    ReconTaskController.ReInitializationResult result2 =
+        taskController.queueReInitializationEvent(
+            ReconTaskReInitializationEvent.ReInitializationReason.MANUAL_TRIGGER);
+    assertEquals(ReconTaskController.ReInitializationResult.SUCCESS, result2,
+        "Second event should be queued successfully after retry delay");
+
+    assertTrue(secondAttempt.await(10, TimeUnit.SECONDS), "Second rebuild should be attempted");
+    Thread.sleep(1000);
     assertEquals(RebuildState.IDLE, NSSummaryTask.getRebuildState(),
         "State should be IDLE after successful rebuild");
   }
 
   /**
-   * Test multiple concurrent attempts - only one should succeed, others rejected.
+   * Test multiple concurrent queueReInitializationEvent() calls.
+   *
+   * This is the KEY test for production behavior - multiple threads
+   * simultaneously calling queueReInitializationEvent(), which is what
+   * actually happens in production (not direct reprocess() calls).
+   *
+   * <p>Important: The queue-based architecture provides SEQUENTIAL processing,
+   * not event deduplication. Multiple successfully queued events will execute
+   * sequentially (not concurrently). The AtomicReference in NSSummaryTask
+   * prevents concurrent execution within a single reprocess() call.
    */
   @Test
-  @Flaky("HDDS-13573")
+  @SuppressWarnings("methodLength")
```

Comment:
Please split method instead of suppressing warning.

(Some code can even be reformatted to reduce line count.)